eclass/qt5-build.eclass

```diff
@@ -694,6 +697,49 @@ qt5_base_configure() {
 
 }
 
+# @FUNCTION: qt5_get_mkspec
+# @INTERNAL:
+# @DESCRIPTION:
+# Returns the right mkspec for the current CHOST/CXX combination
+qt5_get_mkspec() {
+	local spec=
+
+	case ${CHOST} in
```

Comment:
I don't want to duplicate the configure logic in the eclass. The configure already selects the right mkspec most of the time, we just want to force the `*-clang` variant if there's one and the active compiler is clang.

And why do we need to care about all this stuff? aix? solaris? hpux? really? they don't even have a clang mkspec. The only ones we care about that have clang variant are `linux-clang` and `freebsd-clang`.

Also, let's not pass `-platform` at all unless we have to. So with gcc just don't do anything.